yarn build error

### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/antd-reproduce-5x?file=demo.tsx)

### Steps to reproduce

yarn build

### What is expected?

no build error


### What is actually happening?

build error 

| Environment | Info |
| --- | --- |
| antd | 5.21.2 |
| React | react@18.3.1 |
| System | windows |
| Browser | chrome |

---
This is my tsconfig. 
{
  "compilerOptions": {
    "jsx": "react",
    "module": "NodeNext",
    "target": "ESNext",
    "moduleResolution": "NodeNext",
    "outDir": "out",
    "lib": ["ES2022", "DOM"],
    "sourceMap": true,
    "rootDir": "./",
    "strict": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedParameters": true,
    "allowJs": true,
    "types": ["mocha", "node"],
    "resolveJsonModule": true,
    "esModuleInterop": true
  },
  "include": ["src", "test"],
  "exclude": ["node_modules", ".vscode-test"]
}

<!-- generated by ant-design-issue-helper. DO NOT REMOVE -->

yarn build
yarn run v1.22.22
$ run-s build:*
$ tsc
node_modules/antd/es/form/Form.d.ts:5:30 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("scroll-into-view-if-needed")' call instead.

5 import type { Options } from 'scroll-into-view-if-needed';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/antd/es/form/interface.d.ts:2:47 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("scroll-into-view-if-needed")' call instead.

2 export type { Options as ScrollOptions } from 'scroll-into-view-if-needed';
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/antd/es/theme/util/genStyleUtils.d.ts:6:221 - error TS2304: Cannot find name 'CompTokenMap'.        

6     unitless?: (keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<ComponentTokenMap, AliasToken>[C], undefined> extends infer T extends keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<CompTokenMap, AliasToken>[C_1], undefined> ? { [key in T]: boolean; } : never) | undefined;
  
                                                                                                             ~~~~~~~~~~~~

node_modules/antd/es/theme/util/genStyleUtils.d.ts:6:247 - error TS2304: Cannot find name 'C_1'.

6     unitless?: (keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<ComponentTokenMap, AliasToken>[C], undefined> extends infer T extends keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<CompTokenMap, AliasToken>[C_1], undefined> ? { [key in T]: boolean; } : never) | undefined;
  

                      ~~~

node_modules/antd/es/theme/util/genStyleUtils.d.ts:17:225 - error TS2304: Cannot find name 'CompTokenMap'.       

17     unitless?: (keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<ComponentTokenMap, AliasToken>[C_2], undefined> extends infer T_2 extends keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<CompTokenMap, AliasToken>[C_2_1], undefined> ? { [key_2 in T_2]: boolean; } : never) | undefined;
   

 ~~~~~~~~~~~~

node_modules/antd/es/theme/util/genStyleUtils.d.ts:17:251 - error TS2552: Cannot find name 'C_2_1'. Did you mean 
'C_2'?

17     unitless?: (keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<ComponentTokenMap, AliasToken>[C_2], undefined> extends infer T_2 extends keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<CompTokenMap, AliasToken>[C_2_1], undefined> ? { [key_2 in T_2]: boolean; } : never) | undefined;
   

                           ~~~~~

node_modules/antd/es/theme/util/genStyleUtils.d.ts:25:225 - error TS2304: Cannot find name 'CompTokenMap'.       

25     unitless?: (keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<ComponentTokenMap, AliasToken>[C_1], undefined> extends infer T_1 extends keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<CompTokenMap, AliasToken>[C_1_1], undefined> ? { [key_1 in T_1]: boolean; } : never) | undefined;
   

 ~~~~~~~~~~~~

node_modules/antd/es/theme/util/genStyleUtils.d.ts:25:251 - error TS2304: Cannot find name 'C_1_1'.

25     unitless?: (keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<ComponentTokenMap, AliasToken>[C_1], undefined> extends infer T_1 extends keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<CompTokenMap, AliasToken>[C_1_1], undefined> ? { [key_1 in T_1]: boolean; } : never) | undefined;
   

                           ~~~~~

node_modules/antd/es/theme/util/genStyleUtils.d.ts:25:251 - error TS2552: Cannot find name 'C_1_1'. Did you mean 
'C_1'?

25     unitless?: (keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<ComponentTokenMap, AliasToken>[C_1], undefined> extends infer T_1 extends keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<CompTokenMap, AliasToken>[C_1_1], undefined> ? { [key_1 in T_1]: boolean; } : never) | undefined;
   

                           ~~~~~

node_modules/rc-picker/lib/PickerPanel/index.d.ts:40:18 - error TS2430: Interface 'SinglePickerPanelProps<DateType>' incorrectly extends interface 'BasePickerPanelProps<DateType>'.
  Types of property 'defaultValue' are incompatible.
    Type 'DateType | null | undefined' is not assignable to type 'DateType | undefined'.
      Type 'null' is not assignable to type 'DateType | undefined'.

40 export interface SinglePickerPanelProps<DateType extends object = any> extends BasePickerPanelProps<DateType> 
{
                    ~~~~~~~~~~~~~~~~~~~~~~


Found 10 errors in 4 files.

Errors  Files
     1  node_modules/antd/es/form/Form.d.ts:5
     1  node_modules/antd/es/form/interface.d.ts:2
     7  node_modules/antd/es/theme/util/genStyleUtils.d.ts:6
     1  node_modules/rc-picker/lib/PickerPanel/index.d.ts:40
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "build:tsc" exited with 2.
error Command failed with exit code 1.

還是有幾個其他的ERROR， for type checking in d.ts
yarn run v1.22.22
$ run-s build:*
$ tsc
node_modules/antd/es/theme/util/genStyleUtils.d.ts:6:221 - error TS2304: Cannot find name 'CompTokenMap'.

6     unitless?: (keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<ComponentTokenMap, AliasToken>[C], undefined> extends infer T extends keyof Exclude<import("@ant-design/cssinjs-utils").OverrideTokenMap<CompTokenMap, AliasToken>[C_1], undefined> ? { [key in T]: boolean; } : never) | undefined;

--- a/package.json
+++ b/package.json
@@ -107,7 +107,7 @@
   "dependencies": {
     "@ant-design/colors": "^7.1.0",
     "@ant-design/cssinjs": "^1.21.1",
-    "@ant-design/cssinjs-utils": "^1.1.0",
+    "@ant-design/cssinjs-utils": "^1.1.1",
     "@ant-design/icons": "^5.5.1",
     "@ant-design/react-slick": "~1.1.2",
     "@babel/runtime": "^7.25.6",

